lib/src/validators/email_validator.dart

```diff
@@ -5,18 +5,18 @@
 import 'package:reactive_forms/reactive_forms.dart';
 
 /// Validator that requires the control's value pass an email validation test.
-class EmailValidator extends Validator<dynamic> {
+class EmailValidator extends Validator<String> {
   static final RegExp emailRegex = new RegExp(
       r'^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$');
 
   @override
-  Map<String, dynamic> validate(AbstractControl<dynamic> control) {
+  Map<String, dynamic> validate(AbstractControl<String> control) {
     final error = {ValidationMessage.email: control.value};
 
     // don't validate empty values to allow optional controls
     return (control.isNull ||
-            control.value.toString().isEmpty ||
-            emailRegex.hasMatch(control.value.toString().toString()))
+            control.value.isEmpty ||
+            emailRegex.hasMatch(control.value.toString()))
         ? null
         : error;
   }
```

Comment:
This will not allow creating a control with dynamic type as:
```dart
final control = FormControl(
   value: "I'm a String control",
   validators: [Validators.email], // exception is thrown
);
```

Or even create a control with a different custom type like:
```dart
final control = FormControl<CustomEmail>(
   value: CustomEmail("johndoe@gmail.com"),
   validators: [Validators.email], // exception is thrown
);
```
So I suggest leaving the Email validator as **dynamic**